test/cql-pytest/test_permissions.py

```diff
@@ -218,6 +223,14 @@ def test_drop_udf_with_same_name(cql):
         with new_user(cql) as username:
             with new_session(cql, username) as user_session:
                 grant(cql, 'DROP', f'FUNCTION {keyspace}.{fun}(int)', username)
+                # When trying to drop a function without a signature but
+                # there is more than one function with that name, we should
+                # get an InvalidRequest, as is confirmed by the test
+                # test_udf.py::test_drop_udf_with_same_name.
+                # Cassandra fails here because it has a bug: It decided to
+                # complain about the true (but irrelevant) fact that one of
+                # the two functions doesn't have drop permissions and throws
+                # Unauthorized instead of InvalidRequest.
```

Comment:
I assume Cassandra behaves like this only if 2 or 1 out of 2 functions don't have drop permissions? In the latter scenario, if you change which function does have drop permissions, does Cassandra still behave the same way? Which would mean it always first checks drop permissions?

Reply:
I didn't investigate this bug too deeply on the Cassandra side. I'll leave it to them :-)
I should also note that I didn't write this test file originally (Piotr Sarna did), so I can't vouch for how comprehensive it is and maybe it could have been more comprehensive and test more strange scenarios. But I don't want to add any of those in this PR, which I just wanted to fix the test/cql-pytest/run-cassandra failure that annoyed me.